app/models/dataset.rb

```diff
@@ -78,6 +78,46 @@ def as_indexed_json(_options={})
     )
   end
 
+  # map the dataset.as_json so that the keys are in a format that ckan recognises
+  def ckanify_metadata
+    publish_json = self.as_json
+    organisation = Organisation.find(publish_json["organisation_id"])
+    ckan_json = {
+      id: publish_json['uuid'],
+      name: publish_json['name'],
+      title: publish_json['title'],
+      notes: publish_json['summary'],
+      description: publish_json['summary'],
+      organization: {name: organisation.name},
+      update_frequency: convert_freq_to_legacy_format(publish_json['frequency']),
```

Comment:
This needs to be sent as an 'extra' (a key value that is used to extend the schema).  Although extras are promoted to the top level in package_show the same isn't true in reverse...

```
extras: [{key: "update_frequency", value: "...." }]
```

You may find others (like unpublished) may be the same but unpublished in particular is probably unnecessary for new system.

Reply:
Great, thanks Ross, that makes sense.